cmd/cli/serve/util.go

```diff
@@ -2,260 +2,164 @@ package serve
-		S3PreSignedURLDisabled:         cfg.Node.Requester.StorageProvider.S3.PreSignedURLDisabled,
-		TranslationEnabled:             cfg.Node.Requester.TranslationEnabled,
-		JobStore:                       jobStore,
-		DefaultPublisher:               cfg.Node.Requester.DefaultPublisher,
-		NodeInfoStoreTTL:               time.Duration(cfg.Node.Requester.NodeInfoStoreTTL),
-	})
+		EvalBrokerVisibilityTimeout: time.Duration(cfg.Orchestrator.EvaluationBroker.VisibilityTimeout),
+		EvalBrokerMaxRetryCount:     cfg.Orchestrator.EvaluationBroker.MaxRetryCount,
+		WorkerCount:                 cfg.Orchestrator.Scheduler.WorkerCount,
+		TranslationEnabled:          cfg.FeatureFlags.ExecTranslation,
+		JobStore:                    jobStore,
+		DefaultPublisher:            cfg.DefaultPublisher.Type,
+	}
+
+	if cfg.Publishers.Enabled(models.StorageSourceS3) {
+		if cfg.Publishers.S3.Installed() {
+			params.S3PreSignedURLExpiration = time.Duration(cfg.Publishers.S3.PreSignedURLExpiration)
+			params.S3PreSignedURLDisabled = cfg.Publishers.S3.PreSignedURLDisabled
+		}
+	}
+
+	requesterConfig, err := node.NewRequesterConfigWith(params)
 	if err != nil {
 		return node.RequesterConfig{}, err
 	}
 
-	if cfg.Node.Requester.ManualNodeApproval {
+	requesterConfig.DefaultApprovalState = models.NodeMembership.APPROVED
+	if cfg.Orchestrator.NodeManager.ManualApproval {
 		requesterConfig.DefaultApprovalState = models.NodeMembership.PENDING
-	} else {
-		requesterConfig.DefaultApprovalState = models.NodeMembership.APPROVED
 	}
 
 	return requesterConfig, nil
 }
 
-func getNodeType(cfg types.BacalhauConfig) (requester, compute bool, err error) {
-	requester = false
-	compute = false
-	err = nil
-
-	for _, nodeType := range cfg.Node.Type {
-		if nodeType == "compute" {
-			compute = true
-		} else if nodeType == "requester" {
-			requester = true
-		} else {
-			err = fmt.Errorf("invalid node type %s. Only compute and requester values are supported", nodeType)
-		}
+func getNetworkConfig(cfg types.Bacalhau) (node.NetworkConfig, error) {
+	storeDir, err := cfg.NetworkTransportDir()
+	if err != nil {
+		return node.NetworkConfig{}, err
 	}
-	return
-}
-
-func getNetworkConfig(networkStorePath string, cfg types.NetworkConfig) (node.NetworkConfig, error) {
 	return node.NetworkConfig{
-		Port:                     cfg.Port,
-		AdvertisedAddress:        cfg.AdvertisedAddress,
-		Orchestrators:            cfg.Orchestrators,
-		StoreDir:                 networkStorePath,
-		AuthSecret:               cfg.AuthSecret,
-		ClusterName:              cfg.Cluster.Name,
-		ClusterPort:              cfg.Cluster.Port,
-		ClusterAdvertisedAddress: cfg.Cluster.AdvertisedAddress,
-		ClusterPeers:             cfg.Cluster.Peers,
+		Port:                     cfg.Orchestrator.Port,
+		AdvertisedAddress:        cfg.Orchestrator.Advertise,
+		Orchestrators:            cfg.Compute.Orchestrators,
+		StoreDir:                 storeDir,
+		AuthSecret:               cfg.Orchestrator.AuthSecret,
```

Comment:
> There are different Authorization blocks in the orchestrator and compute configs.

Right, in the initial design doc these were mentioned, but not defined. e.g. `Authorization: {}`

> the proposal was to move away from AuthSecret to an Authorization block that can define Secret today and in the future can hold more

That was never proposed in the initial design, but I can see the benefit of such a change. Would you like me to make that change in this PR? i.e. define an `Authorization` block in the Orchestrator and Compute config that contains a single string field called `secret`?

Reply:
> Second issue is we shouldn't create storeDir for compute nodes. Only for orchestrators

I agree, but assumed changing this behavior was outside the scope of this PR as the issue is related to the node setup code rather than config. For now the change could be to only call `cfg.NetworkTransportDir()` if the node is an orchestrator node, otherwise leave the `StoreDir` field of the `NetworkConfig` blank. Does that sounds acceptable to you?

Reply:
I am happy to break the changes into multiple PRs if that makes things go faster, but lets not introduce bugs. The blocker here is we shouldn't use the orchestrator's `AuthSecret` when spinning up a compute node

Reply:
Can you help me understand the bug being _introduced_ here, if any? The behavior of this method should be identical to the behavior prior to this change: https://github.com/bacalhau-project/bacalhau/blob/ae16c2d7383a2a98d9f37da898e27efb5f32bae8/cmd/cli/serve/util.go#L169 unless I am missing something? The auth secret is then only used if the node type is an orchestrator node: https://github.com/bacalhau-project/bacalhau/blob/ae16c2d7383a2a98d9f37da898e27efb5f32bae8/pkg/nats/transport/nats.go#L104

I understand its a bit hard to follow as this is done via bacalhauConfig -> NetworkConfig -> NATSTransportConfig to NATS (server) Options.

Reply:
The bug is using an orchestrator config `cfg.Orchestrator.AuthSecret` to spin a compute node. Also today the `AuthSecret` is not only used by the orchestrator. It is also used by the compute node to set the client token https://github.com/bacalhau-project/bacalhau/blob/e7c052c0425d9688dee47cbcd5fe7542058476dd/pkg/nats/transport/nats.go#L207-L209

Reply:
At present this change is inline with existing behavior, do we agree on that?
If it's not please correct me where I am wrong - I am not attempting to fix any bugs that already exist here, just aiming to keep behavior the same.

My understanding of the `node.NetwokConfig` `AuthSecret` field is that it's meant to be used by both the client (Compute) and the server (Orchestrator): https://github.com/bacalhau-project/bacalhau/blob/main/pkg/node/config_network.go#L12-L15. The configuration is dual-purpose, the server uses it to authenticate clients, while the client uses it to authenticate itself when connecting to the server. 

Now say the `Compute` and `Orchestrator` configs both contain an `Authorization` block containing a `Secret` - similar to what you've described. Which value should we be using for the `AuthSecret`? Further, observe the NATs client is used by both the compute node and the orchestrator node:
- Orchestrator: https://github.com/bacalhau-project/bacalhau/blob/main/pkg/node/requester.go#L290
- Compute: https://github.com/bacalhau-project/bacalhau/blob/main/pkg/node/node.go#L223

Which seems to imply the Server secret must match the client secret when creating an orchestrator node. 

At a high level I believe there is a problem with the overall design here, and I question if a single transport instance should be shared by both the Compute and Orchestrator services when starting a node. But addressing this issue is outside the scope of this PR.